app/assets/javascripts/index.js.erb

```diff
@@ -0,0 +1,666 @@
+// ESLINT:
+/* global L, moment, $ */
+
+'use strict';
+
+const MAX_RADIUS = 50;	// Radius distance in KM
+
+/**
+ * Unit conversions
+ */
+const Unit = {
+	mi2km: mi => mi * 1.609,
+	mi2m: mi => Unit.mi2km(mi) * 1000,
+	km2mi: km => km / 1.609,
+	m2mi: m => Unit.km2mi(m / 1000)
+};
+
+/**
+ * AJAX Wrapper Class
+ */
+class API {
+	/**
+	 * Sends a GET request to url with data
+	 * @param {string} url
+	 * @param {Object} data
+	 * @returns {Promise<Object>} response data
+	 */
+	static get(url, data) {
+		const request = {
+			method: 'GET',
+			url,
+			data
+		};
+
+		return new Promise(resolve => {
+			$.ajax(request).done(data => resolve(data));
+		});
+	}
+}
+
+/**
+ * Wrapper class to help distinguish locations
+ */
+class Location {
+	/**
+	 * Formatted location data
+	 * @param {Object} location - Raw location data
+	 */
+	constructor(location) {
+		Object.assign(this, JSON.parse(JSON.stringify(location)));
+
+		// TODO

```

Comment:
Generally, TODOs should have some description of what should be done, or perhaps a link to the relevant story. Please make sure any TODOs are either removed or explained before your final commit this semester.